client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -145,10 +208,19 @@ function mediaButton( editor ) {
 			return;
 		}
 
+		const dirtyImage = MediaStore.getDirtyImage();
+		let numOfImagesToUpdate = 0;
+
 		isVisualEditMode = ! editor.isHidden();
 
 		if ( isVisualEditMode ) {
-			images = editor.dom.select( 'img' );
+			if ( dirtyImage ) {
+				images = editor.dom.select( `img.wp-image-${ dirtyImage.ID }` );
```

Comment:
Unless there's a huge perf difference, I think it's fine to keep it as it was and iterate over each.